File: dotnet/Kata/External/CurrencyIsoCode.cs
using System.ComponentModel;
using System.Reflection;

namespace Kata.External;

public enum CurrencyIsoCode
{
    [Description("USD")]
    USD,
    [Description("GBP")]
    GBP,
    [Description("EUR")]
    EUR,
    [Description("JPY")]
    JPY
}

static class CurrencyIsoCodeExtensions
{
    public static string GetDescription(this CurrencyIsoCode value)
    {
        FieldInfo field = value.GetType().GetField(value.ToString())!;

        DescriptionAttribute? attribute
            = Attribute.GetCustomAttribute(field, typeof(DescriptionAttribute))
                as DescriptionAttribute;

        return attribute == null ? value.ToString() : attribute.Description;
    }
}

File: php/src/external/CurrencyIsoCode.php
<?php

namespace kata\external;

enum CurrencyIsoCode
{
    case USD;
    case EUR;
    case GBP;
    case JPY;

    public function toString() {
        return match($this) {
            self::USD => "USD",
            self::EUR => "EUR",
            self::GBP => "GBP",
            self::JPY => "JPY"
        };
    }
}